configuration/config2/EnvironmentNode.cpp

```diff
@@ -0,0 +1,210 @@
+/*##############################################################################
+
+HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems�.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+############################################################################## */
+
+#include "EnvironmentNode.hpp"
+
+void EnvironmentNode::addChild(std::shared_ptr<EnvironmentNode> pNode)
+{
+	m_children.insert(std::make_pair(pNode->getName(), pNode));
+}
+
+
+void EnvironmentNode::addAttribute(const std::string &name, std::shared_ptr<EnvValue> pValue)
+{
+	auto retValue = m_attributes.insert(std::make_pair(name, pValue));
```

Comment:
do we not know the return type? when you start using retValue, it might be easier for reviewers to know the type.